tests/test_wcs.py

```diff
@@ -2440,6 +2440,96 @@ def test_fitswcs():
         wcs = galsim.FitsWCS(file_name, dir=dir, suppress_warning=True)
         assert isinstance(wcs, galsim.AffineTransform)
 
+def check_sphere(ra1, dec1, ra2, dec2, atol=1):
+    # Vectorizing CelestialCoord.distanceTo()
+    # ra/dec in rad
+    # atol in arcsec
+    x1 = np.cos(dec1)*np.cos(ra1)
+    y1 = np.cos(dec1)*np.sin(ra1)
+    z1 = np.sin(dec1)
+    x2 = np.cos(dec2)*np.cos(ra2)
+    y2 = np.cos(dec2)*np.sin(ra2)
+    z2 = np.sin(dec2)
+    dsq = (x1-x2)**2 + (y1-y2)**2 + (z1-z2)**2
+    dist = 2*np.arcsin(0.5*np.sqrt(dsq))
+    w = dsq >= 3.99
+    if np.any(w):
+        cross = np.cross(np.array([x1, y1, z1])[w], np.array([x2, y2, z2])[w])
+        crosssq = cross[0]**2 + cross[1]**2 + cross[2]**2
+        dist[w] = np.pi - np.arcsin(np.sqrt(crossq))
+    dist = np.rad2deg(dist)*3600
+    np.testing.assert_allclose(dist, 0.0, rtol=0.0, atol=atol)
+
+
+@timer
+def test_fittedsipwcs():
+    """Test our ability to construct a WCS from x, y, ra, dec tuples
+    """
+    rng = galsim.UniformDeviate(57721)
+    x = np.empty(100, dtype=float)
+    y = np.empty(100, dtype=float)
+    rng.generate(x)
+    rng.generate(y)
+    x *= 150  # rough range of reference WCS image coords
+    y *= 150
+
+    if __name__ == '__main__':
+        test_tags = all_tags
+    else:
+        test_tags = ['TAN', 'SIP', 'TAN-PV']
+
+    dir = 'fits_files'
+
+    for tag in test_tags:
+        print(tag)
+        file_name, ref_list = references[tag]
+        wcs = galsim.FitsWCS(file_name, dir=dir, suppress_warning=True)
+        if not wcs.isCelestial():
+            continue
+        ra, dec = wcs.xyToradec(x, y, units='rad')
+
+        fittedWCS = galsim.FittedSIPWCS(
+            x[:50], y[:50], ra[:50], dec[:50],
+            order=7
+        )
+        # First check that points used in fit are close to fit function values
+        ra_test, dec_test = fittedWCS.xyToradec(x[:50], y[:50], units='rad')
+        check_sphere(
+            ra[:50], dec[:50], ra_test, dec_test,
+            atol=1 if tag == 'ZPN' else 1e-3
+        )
+        # Now check on some "reserve" points
+        ra_test, dec_test = fittedWCS.xyToradec(x[50:], y[50:], units='rad')
+        check_sphere(
+            ra[50:], dec[50:], ra_test, dec_test,
+            atol=30 if tag == 'ZPN' else 1e-3
+        )
```

Comment:
Added abp values.  The code without these appears somewhat unstable anyway (frequent "Unable to solve for image_pos (max iter reached)" encountered).  Even with the abp vals I occasionally got errors with a largish SIP order: ~7 or so.  The reverse direction is still ~50-100 times slower.  I suspect [this python for loop](https://github.com/GalSim-developers/GalSim/blob/releases/2.2/galsim/fitswcs.py#L1370) vs vectorized and maybe even MKL-d instructions for the forward direction.

Reply:
I don't see an obvious reason why I didn't write the InvertAB function to allow it to do all the x,y values at once.  At the worst, the initial application of the ABP matrices could be done faster doing them all at once, and then have a for loop in the C++ layer to do the refinement separately for each position.  Maybe I'll take this PR as an opportunity to do that, unless you want to give it a go.  50-100x is a lot.

Reply:
I'm on it.  Unless you've already started :)

Reply:
Nope.  You go for it.  :)

Reply:
I managed to vectorize the inversion just using numpy (and horner2d).  horner2d also proved faster than the previous forward-direction SIP/PV code that was there.  So now forward direction is ~10% faster and reverse is only ~3-5x slower instead of 50-100x.  

The numpy inversion is a bit suboptimal in that if just one coordinate requires an extra iteration, then the Newton-Raphson update is repeated for every coordinate.  We could probably chunk the input vector up to mitigate that if desired (or move things back down to c++), but I didn't bother.